packages/models/src/errors.ts

```diff
@@ -98,12 +98,31 @@ const makeProblemLogString = (
   return `- title: ${problem.title} - detail: ${problem.detail} - errors: ${errorsString} - original error: ${originalError}`;
 };
 
+type ProblemBuilderOptions = {
+  /**
+   * If true, allows Problem objects received from downstream services
+   * to be passed through directly. If false, they are treated as internal errors.
+   *
+   * @default true
+   */
+  problemErrorsPassthrough?: boolean;
+  /**
+   * If true, replaces the details of errors that map to HTTP 500, or
+   * the ones that are passed through, with a generic message to avoid leaking internal information.
```

Comment:
Also on passthrough errors we need to hide only the ones that are 500, proposal to make it clearer in the doc 👇 
```suggestion
   * If true, return a generic problem whenever an error is mapped to HTTP 500, hiding the underlying error info. This also applies to 500 errors passing through from downstream services.
```